src/CookingPages.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Recipe } from "./recipe";
+import { Image, Button } from "antd";
+import "antd/dist/antd.min.css";
+import "./Cooking.css";
+
+export default function CookingPages() {
+  const emptyRecipe: Recipe = {
+    id: 0,
+    name: "0",
+    thumbnail: {
+      image: "",
+    },
+    description: "",
+    link: ["0"],
+    metadata: {
+      lastViewed: "0",
+      created: "0",
+      timeToCook: "0",
+    },
+    ingredients: [
+      {
+        ingredient: "",
+        starred: true,
+      },
+    ],
+    steps: [
+      {
+        step: 0,
+        directions: "",
+        image: "",
+      },
+    ],
+  };
+
+  const [cookingRecipe, setCookingRecipe] = useState<Recipe>(emptyRecipe);
+  const [loading, setLoading] = useState(true);
+  const [page, setPage] = useState(0);
+
+  const id = useParams().recipeId;
+  if (!id) {
+    throw new Error("id is undefined");
+  }
+
+  const baseUrl = `${process.env.REACT_APP_API_URL}`;
+  if (!baseUrl || baseUrl === "undefined") {
+    throw new Error("The URL environment variable is undefined/missing");
+  }
+  const cookingUrl = baseUrl + "/recipe/" + id;
+
+  const getRecipe = async () => {
+    const response = await fetch(cookingUrl);
+    const data: Recipe = await response.json();
+    if (!data) {
+      throw new Error(
+        "The fetch call did not return any data matching the Recipe Type"
+      );
+    }
+    setCookingRecipe(data);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    getRecipe();
+  }, []);
+
+  const renderHeader = () => {
```

Comment:
You've got a lot of redundant code here.  Can you put the if logic inside the divs?  IE, inside the step bar, have an if for the previous button and an if for the next button